core/types/transaction.go

```diff
@@ -57,9 +56,6 @@ type Transaction struct {
 	hash atomic.Value
 	size atomic.Value
 	from atomic.Value
-
-	// cache of RollupGasData details to compute the gas the tx takes on L1 for its share of rollup data
-	rollupGas atomic.Value
```

Comment:
We should keep the caching around. Especially if we're going to try per-transaction compression-based L1-cost estimation. Otherwise we'll be re-compressing the transaction data every time we need to inspect the transaction fees.
Within the transaction-pool is a good example of where this caching really helps. Ideally we cache the raw constant data length data, and then re-calculate the fee based on the latest basefee information, as applied within the block / tx-pool.

Reply:
restored, but renamed "RollupDataGas" to "L1DataCost" since we'll be adding other data for the cost function changes in the pipeline. Also undid the TransactionToMessage signature change upon further reflection.